### ISSUE TITLE
TDES: Unclear initial values for IV2 and IV3 in interleaved/pipelined modes

### ISSUE THREAD

**Issue by abkarcher** [usnistgov/ACVP#1203]
hello!

I am working on adding support for TDES-CBCI, OFBI, and CFBPX ciphers to libacvp. The general documentation available for these ciphers is very narrow; the primary resource that I can find access to is SP800-20 which is revoked and extremely specific in its testing parameters. I also do not have access to APIs for testing these ciphers.

the pseudo code for these ciphers' monte carlo tests in the ACVP spec indicates that IV2 and IV3 have initial values; but these are not included in the test cases. SP800-20  would imply these should either be `5555555555555555`/`AAAAAAAAAAAAAAAA` or the `mod2^64` functions of those, but it is hard to follow what exactly should be done without further context.

Might anyone be able to clarify how the initial IVs should be defined for the sake of ACVP testing these functions? Apologies if this is maybe a broader question than the scope of ACVP - it seems some of these ciphers are about as old as I am so finding relevant information can be difficult 😃 

Thank you,
Andrew

**Comment by Kritner** [comment]
The IV2 and IV3 for the interleaved/pipeline modes are calculated using IV1 and differing const values for use for IV2 and IV3.

From: https://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-20.pdf

Example from KAT:

![image](https://user-images.githubusercontent.com/8814983/126181936-314e7239-3504-4c8a-b744-cb6887354636.png)

Example from MCT:

![image](https://user-images.githubusercontent.com/8814983/126182183-e37447c6-ffbe-415d-a578-6e42f38f08a4.png)

Basically given IV1, IV2, and IV3 are calculated from it.

Here's a few examples:

![image](https://user-images.githubusercontent.com/8814983/126186756-0ffa7cfb-abaf-4e75-9318-fa93e3ecfc41.png)

```cs
[Test]
[TestCase("0000000000000000", "5555555555555555", "AAAAAAAAAAAAAAAA")]
[TestCase("1111111111111111", "6666666666666666", "BBBBBBBBBBBBBBBB")]
[TestCase("1234567890ABCDEF", "6789ABCDE6012344", "BCDF01233B567899")]
[TestCase("BEEFCAFEBEEFCAFE", "1445205414452053", "699A75A9699A75A8")]
```

**Comment by abkarcher** [comment]
Hello,

Thank you so much for the clarification and the examples! This is extremely helpful. Closing - 

Thanks again,
Andrew
